v2/cmd/save.go

```diff
@@ -0,0 +1,264 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-licenses/v2/config"
+	"github.com/google/go-licenses/v2/dict"
+	"github.com/google/go-licenses/v2/ghutils"
+	"github.com/google/go-licenses/v2/goutils"
+	"github.com/google/licenseclassifier"
+	"github.com/otiai10/copy"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+)
+
+// saveCmd represents the save command
+var saveCmd = &cobra.Command{
+	Use:   "save",
+	Short: "Save full license text locally",
+	Long:  `Save full license text and source code locally to be compliant.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		config, err := config.Load("")
+		if err != nil {
+			klog.ErrorS(err, "Failed: load config")
+			os.Exit(1)
+		}
+		info, err := loadInfo(config.Module.Csv.Path)
+		if err != nil {
+			klog.ErrorS(err, "Failed: load license info csv")
+			os.Exit(1)
+		}
+		err = complyWithLicenses(info, *config)
```

Comment:
P1

Yes, I was aware this was one of the major differences. Let me have a look at the existing behavior and see how save can be done without csv.

In all my cases, I needed to configure some overrides before actually going to the save step. I am curious why that's different for go-licenses v1. Is it because it doesn't scan the full folder?